nixos/modules/services/web-servers/meguca.nix

```diff
@@ -28,68 +28,122 @@ in with lib; {
+        printed to the `errors.log` file.
+      '';
     };
 
-    listenAddress = mkOption {
-      type = types.nullOr types.str;
-      default = null;
-      example = "127.0.0.1:8000";
-      description = "Listen on a specific IP address and port.";
+    imagerMode = mkOption {
+      type = types.int;
+      default = 0;
+
+      description = ''
+        Sets what functionality this instance of the application server will
+		    handle. The functionality is divided around handling image-related
+		    processing.
+
+		    0: handle image processing and serving and all other functionality
+		    1: handle all functionality except for image processing and serving
+		    2: only handle image processing and serving
+      '';
     };
 
-    cacheSize = mkOption {
-      type = types.nullOr types.int;
-      default = null;
-      example = 256;
-      description = "Cache size in MB.";
+    databaseURL = mkOption {
+      type = types.str;
+      example = "postgres://meguca:meguca@localhost:5432/meguca?sslmode=disable";
+      description = "Database URL to connect to on server start.";
     };
 
-    postgresArgs = mkOption {
+    testDatabaseURL = mkOption {
       type = types.str;
-      example = "user=meguca password=dumbpass dbname=meguca sslmode=disable";
-      description = "Postgresql connection arguments.";
+      example = "postgres://meguca:meguca@localhost:5432/meguca_test?sslmode=disable";
+
+      description = ''
+        Database URL to use during tests.
+
+        This URL only serves as the base. The actual databases are created
+        and dropped during testing automatically with prefixes for each
+        database according to the submodule using this test database for
+        running unit tests. This division allows database-related for each
+        submodule to be run concurrently, overall reducing the runtime of
+        unit tests.
+
+        To allow database creation during tests the role used must have the
+        necessary PostgreSQL permissions. These can be granted by running
+
+          ALTER USER $user_name WITH CREATEDB;
+
+        as the administrator postgres user in the psql shell.
+      '';
     };
 
-    postgresArgsFile = mkOption {
-      type = types.path;
-      default = "/run/keys/meguca-postgres-args";
-      example = "/home/okina/meguca/keys/postgres";
-      description = "Postgresql connection arguments file.";
+    cacheSize = mkOption {
+      type = types.float;
+      default = 128.0;
+      example = 255.9;
+
+      description = ''
+        Size limit of internal cache in MB. Once limit is exceeded, the least
+		    recently used records from the cache will be evicted.
+      '';
     };
 
-    compressTraffic = mkOption {
-      type = types.bool;
-      default = false;
-      description = "Compress all traffic with gzip.";
+    listenAddress = mkOption {
+      type = types.str;
+      default = "127.0.0.1:8000";
```

Comment:
changing the default here looks like a bad idea.

Reply:
Seems like it's just been made explicit from internal default before, lgtm.